### ISSUE TITLE
The `QueryRower` interface does not allow for custom databases to be used

### ISSUE THREAD

**Issue by slnt** [Masterminds/squirrel#218]
The `QueryRower` interface makes it so that custom wrappers of `sql.DB` cannot be used in the `.RunWith(myCustomDB).QueryRow().Scan(...)` way. 

Generally the error looks as follows when I attempt to do so:  

```
cannot QueryRow; Runner is not a QueryRower
```

This is rather frustrating for basically all the reasons I gave in #213 

I think the interface should be changed so that it matches with the Golang sql interface. This could, unfortunately, be a breaking change to the interface, though. I'd expect it would probably need a new minor version of the library to be tagged at least.

```go
// currently the interface is:
type QueryRower interface {
    QueryRow(query string, args ...interface{}) RowScanner
}

// the interface should really be:
type QueryRower interface {
    QueryRow(query string, args ...interface{}) *sql.Row
}
```

**Comment by lann** [comment]
`sql.Row` is a struct with unexported fields, so user code cannot implement that interface properly. I ran into this when I was originally writing squirrel and this is best I could come up with. Suggestions welcome, but note that I do not intend to make backward-incompatible changes until/unless I release a new _major_ version.

**Comment by slnt** [comment]
Hmmm I'll take a stab at implementing something for this. I think it can be done without that breaking change, or at least a workaround added.

**Comment by slnt** [comment]
Fixed by #219
